multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java

```diff
@@ -272,7 +272,7 @@ private boolean shouldUpdateCredentials(CloudServiceInstanceExtended service, Cl
             getStepLogger().debug("Existing service parameters: " + SecureSerialization.toJson(serviceParameters));
             return !Objects.equals(service.getCredentials(), serviceParameters);
         } catch (CloudOperationException e) {
-            if (e.getStatusCode() == HttpStatus.BAD_REQUEST) {
+            if (e.getStatusCode() == HttpStatus.BAD_REQUEST || service.isOptional()) {
                 getStepLogger().warnWithoutProgressMessage(Messages.CANNOT_RETRIEVE_SERVICE_INSTANCE_PARAMETERS, service.getName());
                 // TODO: Optimization (Hack) that should be deprecated at some point. So here is a todo for that.
                 return !MapUtils.isEmpty(service.getCredentials());
```

Comment:
What will happen if the service is optional and has specified service parameters and at this line return code is `true`? 
Is this mean that service parameters update will be triggered? How will be handled service parameters update when service instance doesn't exist?

Reply:
This code won't be executed if a service instance doesn't exist.